### PR TITLE
hoc2021: Update sites for Hour of Code 2021 (soon-hoc)

### PR DESCRIPTION
**Pull request by breville**
The site gets a variety of new content for the upcoming `hoc_mode` of `"soon-hoc"`:

- https://code.org/ shows a new hero.  (Removed support for a bunch of previous homepage heroes.)
- https://hourofcode.org/ gets a new hero, and updated social metadata for title, description, image.
- https://code.org/learn and https://hourofcode.com/learn get new banners.
- https://code.org/hourofcode/overview gets a new lower banner image.

For other values of `hoc_mode`, the site remains the same.

Lots of effort has gone into responsive layout, as always.  Due to the nature of this year's design, with the arrow images to the left and right of content, the current behaviour on narrower displays is to hide those images; alternate behaviour on narrower displays is yet to be designed.

Most of the changes made here for `"soon-hoc"` will also apply for `"actual-hoc"`.

# soon-hoc

### code.org/
Screen Shot 2021-10-11 at 3 05 19 PM

### hourofcode.com/
Screen Shot 2021-10-11 at 3 05 23 PM

### code.org/learn
Screen Shot 2021-10-11 at 3 05 25 PM

### hourofcode.com/learn
Screen Shot 2021-10-11 at 3 05 29 PM

### code.org/hourofcode.com/overview
Screen Shot 2021-10-12 at 9 39 26 AM

# current pre-hoc

### code.org/
Screen Shot 2021-10-11 at 3 14 40 PM

### hourofcode.com/
Screen Shot 2021-10-11 at 3 14 42 PM

### code.org/learn
Screen Shot 2021-10-11 at 3 14 45 PM

### hourofcode.com/learn
Screen Shot 2021-10-11 at 3 14 49 PM

### code.org/hourofcode.com/overview
Screen Shot 2021-10-11 at 3 15 48 PM

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml [hunk 1]
```diff
@@ -52,8 +52,11 @@ social:
     = view :top_hoc_tutorials
   .clear
 
+  - hoc_mode = DCDO.get("hoc_mode", CDO.default_hoc_mode)
+  - hoc_image = hoc_mode == "soon-hoc" ? "/images/hoc-cse-banner.png" : "/images/hoc2020-banner.png"
```
Comment (molly-moen) [inline pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: will we also want this banner for `actual-hoc`?

**Review by molly-moen — APPROVED** [review]
Looks good overall, thanks for removing all the old stuff! Just one question

File: pegasus/sites.v3/code.org/public/hourofcode/overview.haml (hunk 1 above)
Comment (breville) [reply pegasus/sites.v3/code.org/public/hourofcode/overview.haml]: Nice catch, fixed.

**Review by KylieModen — APPROVED** [review]
lgtm!

**Pull request merged**